### PR TITLE
feature flag: add feature flag `type_changing_struct_update` for implementation of RFC #2528

### PR DESCRIPTION
**Pull request by kolharsam**
This PR adds a feature flag: `type_changing_struct_update` with regards to the implementation of RFC https://github.com/rust-lang/rfcs/pull/2528.

The implementation issue can be found here: https://github.com/rust-lang/rust/issues/86618

### CODE REVIEW

**Comment by JohnCSimon** [comment]
Ping from triage: 
@kolharsam - can you please address the broken build?

**Comment by crlf0710** [comment]
@kolharsam Ping from triage, CI is still red here. Mind fixing it?

File: compiler/rustc_feature/src/active.rs [hunk 1]
```diff
@@ -666,6 +666,10 @@ declare_features! (
     /// Allows qualified paths in struct expressions, struct patterns and tuple struct patterns.
     (active, more_qualified_paths, "1.54.0", Some(80080), None),
 
+    /// Allows creation of instances of a struct by moving fields that have
+    /// not changed from prior instances of the same struct (RFC #2528)
+    (active, type_changing_struct_update, "1.55.0", Some(86618), None)
```
Comment (camelid) [inline compiler/rustc_feature/src/active.rs:671]: This just needs a trailing comma.

Comment (estebank) [reply compiler/rustc_feature/src/active.rs:671]: Also, the version number will need to be updated, right?

Comment (camelid) [reply compiler/rustc_feature/src/active.rs:671]: Yes, I believe it should be updated to 1.57.0 (because 1.56.0 is about to be promoted to beta).

**Comment by JohnCSimon** [comment]
Triage: 
 
@kolharsam I'm closing this as inactive, please feel free to reopen the PR when you're ready to continue. Thanks.

**Pull request closed without merging**